### PR TITLE
Arf 83871 Form 21a Step 1

### PR DESCRIPTION
**Pull request by williamphelps13**
## Summary

- Added step pages to align with the 21A V2 Designs flow (Note: there are some aspects that align with V3 Designs that were confirmed with design as mostly set - if you do not see the UI element in V2 check the V3 tab.)
- [Designs](https://www.figma.com/design/2afIGOMII0uRI5ck1dWo1w/Form-21a---Apply-for-Accreditation-(CA-%26-Attorneys)?node-id=262-22141&t=JXSWf514L6nfbf8n-4)
- Added e2e testing
- Accredited Representative Facing Team

## Related issue(s)

- [Merge Form 21a Step 1 (Personal Information)](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/83871)
- [[ARF] Digital 21a](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/83627)

## Testing done

- Navigate from Introduction page through all 7 steps especially step 1 to review and submit
- Ensure the pages align with [designs](https://www.figma.com/design/2afIGOMII0uRI5ck1dWo1w/Form-21a---Apply-for-Accreditation-(CA-%26-Attorneys)?node-id=99-3435&t=W5LREYSL0uiLC764-4)

### After
Screenshot 2024-06-14 at 9 30 31 AM
Screenshot 2024-06-14 at 9 31 01 AM
Screenshot 2024-06-14 at 9 31 19 AM
Screenshot 2024-06-14 at 9 31 57 AM
Screenshot 2024-06-14 at 9 32 22 AM
Screenshot 2024-06-14 at 9 32 49 AM
Screenshot 2024-06-14 at 9 33 04 AM
Screenshot 2024-06-14 at 9 39 25 AM
Screenshot 2024-06-14 at 9 39 55 AM
Screenshot 2024-06-14 at 9 40 23 AM
Note: this does not include the address below the radio label custom UI as this design element could be in flux and is not a pattern in the VA Forms Library
Screenshot 2024-06-14 at 9 40 47 AM

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

**Review by gabezurita — COMMENTED** [review]
Looks good! I'll give a review once it's out of draft 😄

**Review by CBonade — DISMISSED** [review]

**Comment by gabezurita** [comment]
Merge conflicts @williamphelps13 !

File: src/applications/accreditation/21a/tests/21a.cypress.spec.js [hunk 1]
```diff
@@ -3,37 +3,89 @@ import path from 'path';
...
+            },
+          ],
+        },
+      });
+      cy.intercept('POST', formConfig.submitUrl);
     },
 
-    // Skip tests in CI until the form is released.
-    // Remove this setting when the form has a content page in production.
+    // Remove this skip when the form has a content page in prod
```
Comment (gabezurita) [inline src/applications/accreditation/21a/tests/21a.cypress.spec.js:88]: Would be good to prepend this with // TODO:

**Review by gabezurita — DISMISSED** [review]
LGTM! Excellent progress.

Will we perform address validation before sending the request to the backend? We could do this as part of follow-on work, but we should have a decision/ticket made for it.

**Comment by gabezurita** [comment]
Side note: a possible way to fix the code climate issues, if you feel it's worth it:
image

```
// src/applications/accreditation/21a/utils/uiSchemas.js

import { radioSchema, radioUI, titleUI, addressSchema, addressUI } from 'platform/forms-system/src/js/web-component-patterns';

// Utility function for radio fields
export const createRadioField = (title, options) => ({
  uiSchema: {
    ...titleUI(title),
    field: radioUI({
      title,
      labels: options,
    }),
  },
  schema: {
    type: 'object',
    properties: {
      field: radioSchema(Object.keys(options)),
    },
    required: ['field'],
  },
});

// Utility function for address fields
export const createAddressField = (title, labels = {}) => ({
  uiSchema: {
    ...titleUI(title),
    address: addressUI({
      labels,
      omit: ['street3'],
    }),
  },
  schema: {
    type: 'object',
    properties: {
      address: addressSchema({
        omit: ['street3'],
      }),
    },
  },
});
```

```
// src/applications/accreditation/21a/pages/01-personal-information-chapter/communicationAddress.js

import { createRadioField } from '../../utils/uiSchemas';

const communicationAddressOptions = {
  HOME: 'Home',
  WORK: 'Work',
  OTHER: 'Other',
};

const communicationAddress = createRadioField('Communication address', communicationAddressOptions);

export default {
  title: 'Communication address',
  path: 'communication-address',
  uiSchema: communicationAddress.uiSchema,
  schema: communicationAddress.schema,
};

// src/applications/accreditation/21a/pages/01-personal-information-chapter/role.js

import { createRadioField } from '../../utils/uiSchemas';

const roleOptions = {
  ATTORNEY: 'Attorney',
  CLAIMS_AGENT: 'Claims agent',
};

const role = createRadioField('Role', roleOptions);

export default {
  title: 'Role',
  path: 'role',
  uiSchema: role.uiSchema,
  schema: role.schema,
};

// src/applications/accreditation/21a/pages/01-personal-information-chapter/workAddress.js

import { createAddressField } from '../../utils/uiSchemas';

const workAddress = createAddressField('Work address', {
  militaryCheckbox: 'I work on a United States military base outside of the U.S.',
  state: 'State/Province/Region',
});

export default {
  title: 'Work address',
  path: 'work-address',
  uiSchema: workAddress.uiSchema,
  schema: workAddress.schema,
};

// src/applications/accreditation/21a/pages/01-personal-information-chapter/otherAddress.js

import { createAddressField } from '../../utils/uiSchemas';

const otherAddress = createAddressField('Other address', {
  militaryCheckbox: 'I live on a United States military base outside of the U.S.',
  state: 'State/Province/Region',
});

export default {
  title: 'Other address',
  path: 'other-address',
  depends: formData => formData.communicationAddress === 'OTHER',
  uiSchema: otherAddress.uiSchema,
  schema: otherAddress.schema,
};

```

File: src/applications/accreditation/21a/tests/21a.cypress.spec.js [hunk 2]
```diff
@@ -3,37 +3,89 @@ import path from 'path';
...
-
-    // Rename and modify the test data as needed.
+    dataDir: path.join(__dirname, 'fixtures', 'data'),
     dataSets: ['test-data'],
-
     pageHooks: {
       introduction: ({ afterHook }) => {
         afterHook(() => {
-          cy.findAllByText(/start/i, { selector: 'button' })
+          cy.findAllByText('Start your application without signing in')
```
Comment (ddzz) [inline src/applications/accreditation/21a/tests/21a.cypress.spec.js:19]: Minor point: can we use `cy.findByText()` here?

**Review by gabezurita — APPROVED** [review]
Minor changes still LGTM! Thank you for being thorough!!

**Review by CBonade — APPROVED** [review]

**Pull request merged**